frontend/src/lib/attach.js

```diff
@@ -0,0 +1,138 @@
+export const ReadyStateEnum = {
+  CONNECTING: 0,
+  OPEN: 1,
+  CLOSING: 2,
+  CLOSED: 3
+}
+
+export function attach (term, socket, pingIntervalSeconds = 30, bidirectional, buffered) {
+  bidirectional = typeof bidirectional === 'undefined' ? true : bidirectional
+  term.__socket = socket
+
+  term.__flushBuffer = () => {
+    term.write(term.__attachSocketBuffer)
+    term.__attachSocketBuffer = null
+  }
+
+  term.__pushToBuffer = (data) => {
+    if (term.__attachSocketBuffer) {
+      term.__attachSocketBuffer += data
+    } else {
+      term.__attachSocketBuffer = data
+      setTimeout(term.__flushBuffer, 10)
+    }
+  }
+
+  let decoder
+
+  term.__getMessage = function (ev) {
+    if (!decoder) {
+      decoder = new TextDecoder()
+    }
+    if (typeof ev.data === 'object' && ev.data instanceof ArrayBuffer) {
+      const buffer = Buffer.from(ev.data)
+      if (buffer.length > 1) {
+        const channel = buffer[0]
+        const data = decoder.decode(buffer.slice(1))
+        switch (channel) {
+          case 1:
+          case 2:
+            displayData(data)
+            break
+          case 3:
+            window['console']['error'](JSON.parse(data))
```

Comment:
ignore if `status` is success, otherwise -> console.err
and try/catch JSON.parse

Reply:
done